gapis/api/vulkan/primeable_image_data.go

```diff
@@ -258,12 +300,12 @@ func (pi *ipPrimeableByPreinitialization) prime(sb *stateBuilder, srcLayout, dst
 // true, the image data will be collected from the shadow memory of the old
 // state image object, which is on the host accessible space. If fromHostData is
 // false, the image data will be collected from the device memory.
-func (p *imagePrimer) newPrimeableImageData(img VkImage, opaqueBoundRanges []VkImageSubresourceRange, fromHostData bool) (primeableImageData, error) {
+func (p *imagePrimer) newPrimeableImageData(srcImg, dstImg VkImage, opaqueBoundRanges []VkImageSubresourceRange, fromHostData bool) (primeableImageData, error) {
```

Comment:
There is a awkward correlation between
`fromHostData` and `dstImg` which if `fromHostData == false` then `dstImg` is expected to be valid. 

Is there any nicer way we can structure this function for this purpose? The simplest way I can think of is to split it out into 3 functions 

newPrimeableImageDataFromHost(srcImg, ....., <no fromHostData>)
newPrimeableImageDataFromDevice(srcImage, dstImage, opaque, ....)
newPrimeableImageDataInternal(<what we have now>)

at least that way people can call one of the 2 named functions with more succinct parameterss